iroh-store/src/store.rs

```diff
@@ -105,15 +108,44 @@ impl Store {
         })
         .await??;
 
-        Ok(Store {
-            inner: Arc::new(InnerStore {
-                content: db,
-                config,
-                next_id: next_id.into(),
-            }),
-        })
+        let rpc_addr = config.rpc.listen_addr;
+
+        let inner = Arc::new(Mutex::new(InnerStore {
+            content: db,
+            config,
+            next_id: next_id.into(),
+        }));
+
+        let inner_cl = Arc::clone(&inner);
+        tokio::spawn(async move {
+            // TODO: handle error
+            rpc::new(rpc_addr, inner_cl).await.unwrap()
+        });
+        Ok(Store { inner })
     }
 
+    #[tracing::instrument(skip(self, links, blob))]
+    pub async fn put<T: AsRef<[u8]>, L>(&self, cid: Cid, blob: T, links: L) -> Result<()>
+    where
+        L: IntoIterator<Item = Cid>,
+    {
+        self.inner.lock().await.put(cid, blob, links).await
+    }
+
+    #[tracing::instrument(skip(self))]
+    pub async fn get(&self, cid: &Cid) -> Result<Option<DBPinnableSlice<'_>>> {
+        let store = self.inner.lock().await;
+        // store.get(cid).await
```

Comment:
this throws an error:
"returns a value referencing data owned by the current function
`store` is borrowed here"

But this is the same set up as `put` and `get_links`, so I'm confused why it's different.

Reply:
this is an issue because you added the mutex, I don’t think you actually need one, so try removing it